thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detection/algorithm/MergeWrapper.java

```diff
@@ -73,13 +86,24 @@ public int compare(MergedAnomalyResultDTO o1, MergedAnomalyResultDTO o2) {
    * @param startTime the start time
    * @param endTime the end time
    */
-  public MergeWrapper(DataProvider provider, DetectionConfigDTO config, long startTime, long endTime) {
+  public MergeWrapper(DataProvider provider, DetectionConfigDTO config, long startTime, long endTime) throws Exception {
     super(provider, config, startTime, endTime);
 
     this.maxGap = MapUtils.getLongValue(config.getProperties(), "maxGap", 0);
     this.maxDuration = MapUtils.getLongValue(config.getProperties(), "maxDuration", Long.MAX_VALUE);
     this.slice = new AnomalySlice().withStart(startTime).withEnd(endTime).withConfigId(config.getId());
     this.nestedProperties = ConfigUtils.getList(config.getProperties().get(PROP_NESTED));
+
+    if (config.getProperties().containsKey(PROP_BASELINE_PROVIDER)) {
```

Comment:
I'd put this with the derived class. IMO the merger has no business computing these values, unless dealing with legacy code

Reply:
or why not add another layer of abstraction? ... yay!